raw_exec windows: add support for setting the task user (#25496)
hashicorp/nomad: website/content/docs/job-specification/task.mdx

  [Docker][] images specify their own default users. This can only be set
  on Linux platforms, and clients can restrict [which drivers][user_drivers]
  are allowed to run tasks as [certain users][user_denylist].

>   On the Windows platform, in case Nomad is running as [system service][service],
>   for the [`raw_exec`][raw_exec] driver less privileged service user can be
>   specified (example: `NT AUTHORITY\LocalService`, `NT AUTHORITY\NetworkService`).

Reviewer: ```suggestion
  On Windows, when Nomad is running as a [system service][service] for the
  [`raw_exec`][raw_exec] driver, you may specify a less-privileged service user.
  For example, `NT AUTHORITY\LocalService`, `NT AUTHORITY\NetworkService`.
```

Updated for our docs style guide and to match the content in the raw exec page.
